middlewares/authorizeRoles.js

```diff
@@ -0,0 +1,57 @@
+const { ROLES } = require("../constants/users");
+
+const VALID_ROLES = [ROLES.SUPERUSER, ROLES.APPOWNER, ROLES.MEMBER];
+
+/**
+ * Check if the user has authorization based on their role.
+ * @param {Array.<String>} allowedRoles - Allowed roles for API consumption.
+ * @param {Object} userRoles - Roles information of the current user.
+ * @returns {Boolean} - Whether the current user is authorized or not.
+ */
+const userHasPermission = (allowedRoles, userRoles) => {
+  let permission = false;
+  Object.keys(userRoles).forEach((role) => {
```

Comment:
Nit: We can write this a bit shorter after our [Discord discussion here](https://discord.com/channels/673083527624916993/970747548052635679/970748034994540635)

Reply:
Refactored to :
```js
return allowedRoles.some((role) => userRoles[`${role}`] === true);
```